src/packages/auth/functions.ts

```diff
@@ -134,7 +134,7 @@ export async function login(username: string, password: string): Promise<AuthRes
           token: null,
         };
       } else {
-        const { allowed_roles, default_role } = await getUserRoles(username, DEFAULT_ROLE, ALLOWED_ROLES);
+        const { allowed_roles, default_role } = await getUserRoles(username, DEFAULT_ROLE[0], ALLOWED_ROLES);
```

Comment:
I had initially left this new mapping logic out of the user / pass based login flow. I'll work on adding it now.

Reply:
Change has been made. Basically just extracted the sso token from our user / pass login, and then called `ssoLogin` using that token, so that we hit the code path that handles this group -> role mapping logic.